### ISSUE TITLE
App Crash: win32_error, "the volume for a file has been externally altered"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [soul4soul/peerblock#208]
```
What steps will reproduce the problem?
1. Clicking on a shortcut or the program itself to launch the program
2.
3.

What is the expected output? What do you see instead?
The error is attatched via screenshot, once the "Ok" button is pushed 
another error window pops up, once that error window is closed via the "Ok" 
button you are back at desktop with no GUI for peerblock but it runs in the 
backgroun (unknown if its blocking anything or not)


What version of PeerBlock are you using? On what operating system? 32- or
64-bit?

I've used both the stable version and the beta versions of peerblock with 
the same error generation.

Windows 7 Professional Unpirated 64bit edition

Please provide any additional information below.  Make sure to attach
peerblock.log and/or any screenshots that would help explain your problem.

Errors attatched


```

Original issue reported on code.google.com by `ILove24n...@gmail.com` on 20 Nov 2009 at 11:29

Attachments:
- [error.jpg](https://storage.googleapis.com/google-code-attachments/peerblock/issue-208/comment-0/error.jpg)
- [error2.jpg](https://storage.googleapis.com/google-code-attachments/peerblock/issue-208/comment-0/error2.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Not sure why/how, but it appears as though the peerblock.conf file has become
corrupt.  A few things you could do to help us figure this out:

1.  Can you open peerblock.conf (should be in your PeerBlock install directory) 
with
notepad?

2.  Try renaming peerblock.conf to e.g. "peerblock.conf.bad"

Renaming peerblock.conf (or reinstalling PeerBlock and selecting "Reset 
Peerblock's
Settings") I think should resolve the problem, though at a cost of losing your
previous configuration.  If 1. works, it's possible that simply saving the file 
and
overwriting the old one will work, or renaming peerblock.conf as above then 
opening
it in notepad and copying the contents of the file into a new file named 
peerblock.conf.

Also, could you zip and attach your peerblock.conf file so we can see if that 
file
itself is causing our problems?  Your peerblock.log file would be helpful as 
well, to
verify that the .conf is in fact the one Windows is complaining about here...
```

Original comment by `peerbloc...@gmail.com` on 1 Dec 2009 at 5:16
- Changed title: **App Crash: win32_error, "the volume for a file has been externally altered"**
- Changed state: **Accepted**
- Added labels: **Milestone-1.1**, **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Issue 211 has been merged into this issue.
```

Original comment by `peerbloc...@gmail.com` on 1 Dec 2009 at 5:17

**Comment by GoogleCodeExporter** [comment]
```
I have just experienced this problem as well.

Peerblock v. 1.0.0.181 on Win 7 64 bit legit copy.

peerblock.conf had been reduced to a 0 KB file.  Suggested fix above did repair 
the
issue.  

Pertinent sections of the log as well as results of the most recent successful
startup prior to the crash are attached.

Keep up the good work guys, love the software!
```

Original comment by `squeakym...@gmail.com` on 10 Dec 2009 at 9:41

Attachments:
- [errorlog.txt](https://storage.googleapis.com/google-code-attachments/peerblock/issue-208/comment-3/errorlog.txt)

**Comment by GoogleCodeExporter** [comment]
```
Follow-up thought: the last time I had shut down Windows was after it recovered
strangely from sleep.  The mouse did not come back and so I did a three finger 
salute
and reboot.  I am wondering if this might be related to Bug #166 and the conf 
did not
properly write its shutdown?
```

Original comment by `squeakym...@gmail.com` on 10 Dec 2009 at 9:45

**Comment by GoogleCodeExporter** [comment]
```
This particular error has been reported a few times recently, I believe some 
people
said only after installing some recent Windows update - not sure if Windows now
reports this error for different scenarios than it used to, or what.  Either 
way, we
should be able to more gracefully handle this error condition, for example by
initially saving our config to a temporary file then replacing the current 
.conf file
with that temporary one after it's been successfully written.  I have a few 
commits I
need to make to stabilize the current development-version of our code a bit 
more, but
will try to get to doing this and creating a test-version soon.
```

Original comment by `peerbloc...@gmail.com` on 16 Dec 2009 at 2:36

**Comment by GoogleCodeExporter** [comment]
```
Just happened to me using Vista 32.  The conf file was 0 bytes.  I deleted it 
and
Peerblock started working again.
```

Original comment by `Mako...@gmail.com` on 6 Jan 2010 at 1:00

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r253.
```

Original comment by `peerbloc...@gmail.com` on 8 Jan 2010 at 4:39
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
If we detect a peerblock.conf with size of 0 bytes we will now ignore it, and 
run
through the Startup Wizard.  This should at least workaround the issue if it 
occurs.
 This fix will be in the next Beta Release of PeerBlock.

If people start seeing symptoms of this workaround - i.e. the Startup Wizard 
being
run through even though the machine is supposedly working - we can look further 
into
what was causing the problem in the first place.  Issue #166 may in fact help
alleviate this pain, or if might be worth saving aside a copy of your "last 
known
good .conf file" so that we can revert to it if your current one gets corrupt 
for any
reason.
```

Original comment by `peerbloc...@gmail.com` on 8 Jan 2010 at 4:44

**Comment by GoogleCodeExporter** [comment]
```
What steps will reproduce the problem?
1. Start Peerblock
2. (Somehow) have bad 'peerblock.conf' file.
3.

What is the expected output? What do you see instead?
Expect to see program window; error message in log, "WARNING: unknown file 
type".

What version of PeerBlock are you using? On what operating system? 32- or
64-bit?
Version 1.0, Windows XP SP3, 32-bit.

Please provide any additional information below.  Make sure to attach
peerblock.log and/or any screenshots that would help explain your problem.

   This program worked once, I left it active for a few days. Later, I had to reboot because of a botched Java program installation that hung up, I could not clear directly. (Plus XP needs a good swift kick - I mean reboot - every once in a while.)
   Since the reboot I haven't been able to start Peerblock.
   I use AVG free virus protection and the XP firewall; I don't see a connection but wanted to inform you.
   CONF file attached; the bad one. Renaming this and restarting worked fine.

 Relevant portion of log follows:
 =============================================
Logging Started...
[11/17/2010] [08:57:45.750] [05548]  PeerBlock Starting
[11/17/2010] [08:57:45.750] [05548]  PeerBlock 1.0
[11/17/2010] [08:57:45.750] [05548]  Flushed tracelog
[11/17/2010] [08:57:45.750] [05548]  Running on OS: Windows XP 32-bit - 
Build:[2600], SP:[3.0]
[11/17/2010] [08:57:45.750] [05548]  User running as Admin
[11/17/2010] [08:57:45.750] [05548]  Created program mutex
[11/17/2010] [08:57:45.765] [05548]  Initialized common controls
[11/17/2010] [08:57:45.765] [05548]  Registered color picker
[11/17/2010] [08:57:45.765] [05548]  Initialized winsock
[11/17/2010] [08:57:45.765] [05548]  Successfully 
PreventSetUnhandledExceptionFilter()
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [Main_OnInitDialog]  > Entering 
routine.
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [Main_OnInitDialog]    loading 
config
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]  > Entering routine.
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [LoadFile]    loading 
file:[C:\Program Files\PeerBlock\peerblock.conf]
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [LoadFile]    Successfully 
loaded file:[C:\Program Files\PeerBlock\peerblock.conf]
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    found peerblock 
configuration file
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
root element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
settings element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
logging element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
tracelogging element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
colors element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
windowing element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
updates element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
messages element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
ports element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]    parsing config 
lists element
[11/17/2010] [08:57:45.765] [05548]  [Configuration] [Load]  < Leaving routine.
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [Main_OnInitDialog]    Config 
loaded.
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [Main_OnInitDialog]    checking 
if previous-release updates are required
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [PerformPrevRelUpdates]    
Checking for old peerguardian-hosted lists, and updating any found to 
iblocklist.com-hosted ones
[11/17/2010] [08:57:45.765] [05548]  [mainproc] [Main_OnInitDialog]    
resetting g_filter
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(2)]   preparing to load 
driver - name: [IpFilterDriver], file: [System32\Drivers\IpFltDrv.sys]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]   loading driver - 
name: [IpFilterDriver], file: [System32\Drivers\IpFltDrv.sys], devfile: 
[\\.\IpFilterDriver]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    driver not currently 
installed as service
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    service name: 
[System32\DRIVERS\ipfltdrv.sys], file name: [System32\Drivers\IpFltDrv.sys]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    Driver-service is 
running
[11/17/2010] [08:57:45.765] [05548]  [driver] [start]    driver already started
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(1)]   preparing to load 
driver - name: [pbfilter]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(2)]   preparing to load 
driver - name: [pbfilter], file: [C:\Program Files\PeerBlock\pbfilter.sys]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]   loading driver - 
name: [pbfilter], file: [C:\Program Files\PeerBlock\pbfilter.sys], devfile: 
[\\.\pbfilter]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    driver not currently 
installed as service
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    service name: 
[\??\C:\Program Files\PeerBlock\pbfilter.sys], file name: [C:\Program 
Files\PeerBlock\pbfilter.sys]
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    paths don't match, 
removing and recreating driver-service
[11/17/2010] [08:57:45.765] [05548]  [driver] [load(3)]    service state: 
[SERVICE_RUNNING]
[11/17/2010] [08:57:45.765] [05548]  
[11/17/2010] [08:57:46.125] [05548]  [driver] [load(3)]    finished re-creating 
driver-service, now starting it
[11/17/2010] [08:57:46.125] [05548]  [driver] [load(3)]    Driver-service NOT 
running!
[11/17/2010] [08:57:46.125] [05548]  [driver] [load(3)]    - service state: [1]
[11/17/2010] [08:57:46.125] [05548]  [driver] [start]    starting driver
[11/17/2010] [08:57:46.125] [05548]  [driver] [start]   getting handle to 
driver - devfile: [\\.\pbfilter]
[11/17/2010] [08:57:46.125] [05548]  [driver] [start]    started driver
[11/17/2010] [08:57:46.125] [05548]  [pgfilter_base] [start_thread]    creating 
thread_thunk
[11/17/2010] [08:57:46.125] [05548]  [mainproc] [Main_OnInitDialog]    g_filter 
reset.
[11/17/2010] [08:57:46.125] [05548]  [mainproc] [Main_OnInitDialog]    getting 
tabs
[11/17/2010] [08:57:46.125] [05548]  [LogProc] [Log_OnInitDialog]    updating 
status
[11/17/2010] [08:57:46.140] [05548]  [mainproc] [Main_OnInitDialog]    setting 
block
[11/17/2010] [08:57:46.140] [05548]  [mainproc] [Main_OnInitDialog]    setting 
HTTP block
[11/17/2010] [08:57:46.140] [05548]  [mainproc] [Main_OnInitDialog]    loading 
lists
[11/17/2010] [08:57:46.140] [05548]  [LoadLists]  > Entering routine.
[11/17/2010] [08:57:46.140] [05548]  [LoadLists]    generating cache
[11/17/2010] [08:57:46.140] [05548]  [GenCache]  > Entering routine.
[11/17/2010] [08:57:46.140] [05548]  [GenCache]  < Leaving routine (false).
[11/17/2010] [08:57:46.140] [05548]  [LoadLists]    loading lists
[11/17/2010] [08:57:46.140] [05548]  [LoadList]    WARNING: unknown file type

```

Original comment by `Trelli...@gmail.com` on 17 Nov 2010 at 5:19

Attachments:
- [peerblock.conf.bad](https://storage.googleapis.com/google-code-attachments/peerblock/issue-208/comment-9/peerblock.conf.bad)
